WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCouponViewModel.swift

```diff
@@ -22,7 +22,16 @@ final class AddEditCouponViewModel: ObservableObject {
         }
     }
 
-    @Published private(set) var coupon: Coupon?
+    private var coupon: Coupon? {
+        didSet {
+            amountField = coupon?.amount ?? ""
+            codeField = coupon?.code ?? ""
```

Comment:
Nit: How about extracting this to a method to populate data - we may need to add more logic to it such as accepted products, categories and expiry date.

Reply:
Sure it's a good idea, I'll do that when I'll start implementing the core logic 👍